### DISCUSSION TITLE
File Copy to SD Card

### DISCUSSION THREAD

**Discussion by bjs-pdx** [gtxaspec/wz_mini_hacks#227] [Q&A] [answered]
Step 4 of the V2 installation says:

     Copy all the files inside of SD_ROOT to your micro sd card

Inside that directory is a subdirectory wz_mini.  Beneath that there are many other subdirectories.  I made the assumption that I need to copy everything below wz_mini to the root of the sd card.  Is that correct or do I just copy wz_mini to the root?  Instructions are unclear.

**Comment by Bob-AOL** [comment, marked as answer]
Yes, it is implied that there are multiple individual files in SD_ROOT -- which clearly there are not, at least not with this particular revision.
I felt adventurous and just copied the wz_mini folder instead (not the individual files) and it worked fine for me.

**Reply by bjs-pdx** [reply]
I tried that and it appears to have worked.  I shutdown the camera and looked at the logs and there are boot entries.

A couple of additional questions:  Is SSH running on the wifi connection?  The V2 installation instructions also say "You should hear audio prompts from the camera once it has booted successfully."  What sort of audio prompts should I hear?  I didn't hear anything during the setup process.

**Reply by baudneo** [reply]
`cp -r /xxx/SD-ROOT/* /PATH/TO/SDCARD`

**Comment by bjs-pdx** [comment]
Thanks, but already got that part figured out. What about my other questions?

**Reply by Bob-AOL** [reply]
My process involved a V3. I kept it as simple as possible out of the gate. No SSH. 

I did enable the web server to explore access there. I would advise doing so simply for the RTSP configuration template I found included. Not secure, but handy while getting one's feet wet with some of the basic conf variables. Accessible on port 80. (There is also some car based stuff available if you're into the car thing.) 

SSH would be highly recommended once a comfort zone with that aspect of the platform has been achieved. Not an operational requirement, though. 

The V3 did make some noise / illegible comments during the first boot from the SD card. I didn't detect any of it being in the context of prompts or instructions and simply proceeded. Haven't heard it again since.

**Reply by bjs-pdx** [reply]
Okay.  I've got the firmware running on a PanV2 and a Regular V2.  In both instances the web page says they are V3s.

![image](https://user-images.githubusercontent.com/14082937/183765806-26f05fa5-c6de-438e-92ee-f334952c6a58.png)

Also, on the V2, no video image is displayed on the web page.

**Reply by gtxaspec** [reply]
regarding the broken image on T20 devices, this is a known issue, related to libcallback's jpeg routine not working.  Working on a fix.  @virmaior is the wyze v3 text hard coded?

**Reply by virmaior** [reply]
@gtxaspec I'll check. Obviously it shouldn't be hard-coded...

**Reply by gtxaspec** [reply]
maybe we can have it read the model of the device? the detection and storage logic is in `wz_mini/etc/init.d/S04model` 

or

just read the hostname of the device

**Reply by virmaior** [reply]
it was supposed to be reading it from an environment variable but maybe I never got around to it. I'll look at s04model and get this fixed.

**Reply by bjs-pdx** [reply]
Just an update the time seems to be hard coded as well in the web page as it never updates and only shows 7:01:19 PM PDT on 8/7/2022.

**Reply by virmaior** [reply]
@bjs-pdx the time isn't hard-coded ... that's the version of wz_mini as in the version is a date time.

**Reply by bjs-pdx** [reply]
Ah....thank you!  Just confused I guess and just getting started on this.

**Reply by Bob-AOL** [reply]
The date/time had me a little confused, too. I figured it was established either prior to, or when I booted for the first time. Above my pay grade. I'm just delighted with the stuff that works!

I do notice an issue with Blue Iris not recognizing my V3 as supporting Intel HA anymore when booting from the SD card.  Started a new thread on the subject for any input or suggestions.

**Reply by virmaior** [reply]
@gtxaspec pull request in.

**Reply by virmaior** [reply]
@gtxaspec  looking at that output. does a v2 store the firmware version in a different location? On V3 its in `/configs/app.ver`. Is it somewhere else on V2s?

**Reply by gtxaspec** [reply]
/system/bin/app.ver
